usaspending_api/references/management/commands/load_gtas.py

```diff
@@ -72,16 +89,29 @@ def broker_fetch_sql(self):
         """
 
     def column_statements(self):
-        return "\n".join(
-            [
-                f"""COALESCE(SUM(CASE WHEN line IN ({','.join([str(elem) for elem in val])}) THEN sf.amount ELSE 0 END), 0.0) AS {key},"""
-                for key, val in DERIVED_COLUMNS.items()
-            ]
-        ) + "\n".join(
-            [
-                f"""COALESCE(SUM(CASE WHEN line IN ({','.join([str(elem) for elem in val])}) THEN sf.amount * -1 ELSE 0 END), 0.0) AS {key},"""
-                for key, val in INVERTED_DERIVED_COLUMNS.items()
-            ]
+        return (
+            "\n".join(
+                [
+                    f"""COALESCE(SUM(CASE WHEN line IN ({','.join([str(elem) for elem in val])}) THEN sf.amount ELSE 0 END), 0.0) AS {key},"""
+                    for key, val in DERIVED_COLUMNS.items()
+                ]
+            )
+            + "\n".join(
+                [
+                    f"""COALESCE(SUM(CASE WHEN line IN ({','.join([str(elem) for elem in val])}) THEN sf.amount * -1 ELSE 0 END), 0.0) AS {key},"""
+                    for key, val in INVERTED_DERIVED_COLUMNS.items()
+                ]
+            )
+            + "\n".join(
+                [
+                    f"""COALESCE(SUM(CASE
+                        WHEN line IN ({','.join([str(elem) for elem in val["before_year"]])}) AND fiscal_year < {val["change_year"]} THEN sf.amount * -1
+                        WHEN line IN ({','.join([str(elem) for elem in val["year_and_after"]])}) AND fiscal_year >= {val["change_year"]} THEN sf.amount * -1
+                        ELSE 0
+                    END), 0.0) AS {key},"""
+                    for key, val in DERIVED_COLUMNS_DYNAMIC.items()
+                ]
+            )
         )
 
```

Comment:
Do you think splitting this "one-liner" into separate statements is cleaner?

```suggestion
        simple_fields = [
            f"COALESCE(SUM(CASE WHEN line IN ({','.join([str(elem) for elem in val])}) THEN sf.amount ELSE 0 END), 0.0) AS {key},"
            for key, val in DERIVED_COLUMNS.items()
        ]
        inverted_fields = [
            f"COALESCE(SUM(CASE WHEN line IN ({','.join([str(elem) for elem in val])}) THEN sf.amount * -1 ELSE 0 END), 0.0) AS {key},"
            for key, val in INVERTED_DERIVED_COLUMNS.items()
        ]
        year_specific_fields = [
            f"""COALESCE(SUM(CASE
                        WHEN line IN ({','.join([str(elem) for elem in val["before_year"]])}) AND fiscal_year < {val["change_year"]} THEN sf.amount * -1
                        WHEN line IN ({','.join([str(elem) for elem in val["year_and_after"]])}) AND fiscal_year >= {val["change_year"]} THEN sf.amount * -1
                        ELSE 0
                    END), 0.0) AS {key},"""
            for key, val in DERIVED_COLUMNS_DYNAMIC.items()
        ]
        return "\n".join(simple_fields + inverted_fields + year_specific_fields)
```

Reply:
Yes, I am good with this change. It does look cleaner